File: app/src/main/java/com/kobrakid/retroachievements/database/GameDao.java
package com.kobrakid.retroachievements.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface GameDao {

    @Query("SELECT * FROM game")
    List<Game> getGameList();

    @Query("SELECT * FROM game WHERE id == :gameID")
    List<Game> getGameWithID(int gameID);

    @Query("DELETE FROM game")
    void clearTable();

    @Insert
    void insertGame(Game game);

    @Update
    void updateGame(Game game);

    @Delete
    void deleteGame(Game game);

}

File: app/src/main/java/com/kobrakid/retroachievements/database/GameDao.kt
package com.kobrakid.retroachievements.database

import androidx.room.*

@Dao
interface GameDao {
    @get:Query("SELECT * FROM game")
    val gameList: List<Game>

    @get:Query("SELECT * FROM game WHERE NumAchievements > 0 AND (NumAchievements == NumAwardedToUser OR NumAchievements == NumAwardedToUserHardcore)")
    val masteredGames: List<Game>

    @Query("SELECT * FROM game WHERE id == :gameID")
    fun getGameWithID(gameID: String): List<Game>

    @Query("SELECT * FROM game WHERE ConsoleID == :consoleID")
    fun getGamesFromConsoleByID(consoleID: String): List<Game>

    @Transaction
    @Query("SELECT * FROM game")
    fun getGamesWithAchievements(): List<GameWithAchievements>

    @Query("DELETE FROM game")
    fun clearTable()

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertGame(game: Game?)

    @Update
    fun updateGame(game: Game?)

    @Delete
    fun deleteGame(game: Game?)
}